private/bufpkg/bufplugin/bufplugindocker/docker.go

```diff
@@ -225,6 +225,11 @@ func NewClient(logger *zap.Logger, cliVersion string, options ...ClientOption) (
 		client.WithHTTPHeaders(map[string]string{
 			"User-Agent": BufUpstreamClientUserAgentPrefix + cliVersion,
 		}),
+		// NOTE: This setting is not safe when a client is called from multiple goroutines:
+		// - https://github.com/moby/moby/issues/43729#issuecomment-1174446640
+		//
+		// Our usage is for the CLI with a single goroutine so we're ok to enable for compatibility.
+		client.WithAPIVersionNegotiation(),
```

Comment:
Are we sure we want to bake this into the CLI?

Reply:
unfortunately it cannot be set from the [env](https://pkg.go.dev/github.com/docker/docker/client#FromEnv) like most other options

Reply:
i guess you could set it manually `DOCKER_API_VERSION=1.x`, but that's obviously not great (nor recommended).

this seems reasonable, I just hope the negotiation isn't buggy.

Reply:
The updated code will skip setting the version if you've provided an override via the env var:

https://github.com/moby/moby/blob/d7573ab8672555762688f4c7ab8cc69ae8ec1a47/client/client.go#L227-L244